### PR TITLE
divesite.c: fix double free() and small cleanup

### PR DESCRIPTION
**Pull request by neolit123**
### Describe the pull request:

- [x] Bug fix
- [ ] Functional change
- [ ] New feature
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

prevent an eventual crash when calling clear_dive_site(), due to a double free(). OS dependent, stdlib implementation dependent, undefined behaviour...

we might have caught this with static analysis.

### Changes made:
1) remove double free
2) use NULL instead of 0 for clarity

### Related issues:
none?

### Additional information:
none

### Mentions:
@dirkhh

### CODE REVIEW

File: core/divesite.c [hunk 1]
```diff
@@ -314,7 +314,6 @@ void clear_dive_site(struct dive_site *ds)
 	free(ds->name);
 	free(ds->notes);
 	free(ds->description);
-	free(ds->name);
```
Comment (atdotde) [inline core/divesite.c:317]: Are you sure this should be deleted and not replaced by free(ds->country)?

Comment (neolit123) [reply core/divesite.c:317]: i think one of the later changes had moved `ds->country` to be part of the taxonomy.

Comment (neolit123) [reply core/divesite.c:317]: yep, here is the latest `struct dive_site`:
https://github.com/Subsurface-divelog/subsurface/blob/master/core/divesite.h#L16

Comment (dirkhh) [reply core/divesite.c:317]: @atdotde There is no ds->country anymore...

**Comment by dirkhh** [comment]
LGTM

**Pull request merged**